sympy/sets/sets.py

```diff
@@ -994,6 +993,29 @@ def _eval_imageset(self, f):
             return
         var = f.variables[0]
 
+        if expr.is_Piecewise:
+            result = S.EmptySet
+            domain_set = self
+            for i in expr.args:
```

Comment:
`i` is almost always used as a numerical index, you can rather use something like `for (p_expr, p_cond) in expr.args:` I appended `p_` to signify that is this is a piecewise expression, ofcourse you can use something more informative.